infrastructure.tf

```diff
@@ -194,3 +194,229 @@ resource "aws_iam_role_policy" "gcp_to_cwl" {
 }
 EOF
 }
+
+////
+// Firehose CWL Log processor
+//
+
+resource "aws_iam_role" "firehose_processor" {
+  name               = "firehose-cwl-log-processor-staging"
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Principal": {
+        "Service": "lambda.amazonaws.com"
+      },
+      "Action": "sts:AssumeRole"
+    }
+  ]
+}
+EOF
+}
+
+resource "aws_iam_role_policy" "firehose_processor" {
+  name   = "firehose-cwl-log-processor-staging"
+  role   = "firehose-cwl-log-processor-staging"
+  policy = <<EOF
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Action": [
+                "logs:CreateLogStream",
+                "logs:PutLogEvents"
+            ],
+            "Resource": "arn:aws:logs:*:*:*"
+        },
+        {
+            "Effect": "Allow",
+            "Action": "logs:CreateLogGroup",
+            "Resource": "arn:aws:logs:*:*:*"
+        },
+        {
+            "Effect": "Allow",
+            "Action": "firehose:*",
+            "Resource": "arn:aws:firehose:*:*:*"
+        }
+    ]
+}
+EOF
+}
+
+////
+// Firehose to ES
+//
+
+resource "aws_iam_role" "kinesis-firehose-es" {
+  name               = "kinesis-firehose-es-staging"
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Principal": {
+        "Service": "firehose.amazonaws.com"
+      },
+      "Action": "sts:AssumeRole"
+    }
+  ]
+}
+EOF
+}
+
+resource "aws_iam_role_policy" "kinesis-firehose-es" {
+  name   = "kinesis-firehose-es-staging"
+  role   = "kinesis-firehose-es-staging"
+  policy = <<EOF
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Action": [
+                "logs:CreateLogStream",
+                "logs:DescribeLogStreams",
+                "logs:PutLogEvents"
+            ],
+            "Resource": "arn:aws:logs:*:*:*"
+        },
+        {
+            "Effect": "Allow",
+            "Action": "logs:CreateLogGroup",
+            "Resource": "*"
+        },
+        {
+            "Effect": "Allow",
+            "Action": "es:*",
+            "Resource": "arn:aws:es:*:*:*"
+        },
+        {
+            "Effect": "Allow",
+            "Action": "s3:*",
+            "Resource": "arn:aws:s3:::*"
+        },
+        {
+       "Sid": "UseLambdaFunction",
+       "Effect": "Allow",
+       "Action": [
+           "lambda:InvokeFunction",
+           "lambda:GetFunctionConfiguration"
+       ],
+       "Resource": "arn:aws:lambda:*:*:*"
```

Comment:
If possible, it would be good to limit the invocation permission just to the relevant function.

Reply:
agreed, will update